packages/block-editor/src/components/inserter/test/menu.js

```diff
@@ -91,12 +91,30 @@ const items = [
 	reusableItem,
 ];
 
+const categories = [
+	{ slug: 'common', title: 'Common Blocks', icon: null },
+	{ slug: 'formatting', title: 'Formatting', icon: null },
+	{ slug: 'layout', title: 'Layout Elements', icon: null },
+	{ slug: 'widgets', title: 'Widgets', icon: null },
+	{ slug: 'embed', title: 'Embeds', icon: null },
+	{ slug: 'reusable', title: 'Reusable Blocks', icon: null },
+];
+
+const collections = {
```

Comment:
What's the reason that collections use a different structure than categories?

In general, it seems to work fundamentally different.

Reply:
The way we access collections makes it easier to do so using an object than an array, so this makes the code simpler. I can change it to be an array if you think that matching the structure for categories is better.

Reply:
I'm more concerned about the reducer and type of actions used. I don't remember how categories are managed by plugins. I would need to find some snippets to do the comparison. Ideally, we provide a way to manage it in a similar way on a higher level.

Reply:
Here's how Jetpack does it: https://github.com/Automattic/jetpack/blob/4c63a49fc0139c3296248e09ee69dedb537e024f/extensions/shared/block-category.js